src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ServiceConnection.cs

```diff
@@ -67,56 +70,67 @@ protected override Task CleanupConnections()
             return Task.CompletedTask;
         }
 
-        protected override Task OnConnectedAsync(OpenConnectionMessage openConnectionMessage)
+        protected override async Task OnConnectedAsync(OpenConnectionMessage openConnectionMessage)
         {
-            // Writing from the application to the service
-            _ = OnConnectedAsyncCore(openConnectionMessage);
+            // Create channel per client to async process message
+            var connectionId = openConnectionMessage.ConnectionId;
+            var channel = Channel.CreateUnbounded<ServiceMessage>();
```

Comment:
I'm not sure we should be using an unbounded channel. We may want to make sure we're not flooding the channels unexpectedly. Use a bounded channel instead of some reasonable size. We can use that as a metric if things are backed up on the server (the ability to look at the queue length)

Reply:
Shall we make this configurable in ServiceOptions since it may depend on customer's app server? What default size is suitable? And what shall we do if it exceeds? And if there're a lot messages pending waiting for the channel available, will this stuck ping message?

Reply:
Spoke to @vicancy about this and we're going to open an issue and leave it unbounded for now. We'll need to address this behavior in the .NET Core version of the SDK as well.

>  And if there're a lot messages pending waiting for the channel available, will this stuck ping message?

No it won't be stuck. We could make it drop messages.

Reply:
OK.